### PR TITLE
HPCC-21909 Allow the maximum search to be specified for EditDistance

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review

File: ecllibrary/std/Str.ecl [hunk 1]
```diff
@@ -305,19 +305,21 @@ EXPORT STRING CombineWords(SET OF STRING words, STRING separator) := lib_stringl
  * 
  * @param _left         The first string to be compared.
  * @param _right        The second string to be compared.
- * @return              The minimum edit distance between the two strings.
+ * @param radius        The maximum edit distance that is acceptable, or 0 for no limit.  Defaults to 0.
+ * @return              The minimum edit distance between the two strings.  Edit distances above radius will
+                        return a value larger than radius.
```
Comment (richardkchapman) [inline ecllibrary/std/Str.ecl]: perhaps should say "an arbitrary value larger than radius"

File: plugins/unicodelib/unicodelib.cpp [hunk 2]
```diff
@@ -1767,6 +1768,26 @@ UNICODELIB_API unsigned UNICODELIB_CALL ulUnicodeLocaleEditDistance(unsigned lef
 }
 
 
+UNICODELIB_API unsigned UNICODELIB_CALL ulUnicodeLocaleEditDistanceV2(unsigned leftLen, UChar const * left, unsigned rightLen, UChar const * right, char const * localename, unsigned radius)
+{
+    BreakIterator* bi = 0;
```
Comment (richardkchapman) [inline plugins/unicodelib/unicodelib.cpp]: Should existing function ulUnicodeLocaleEditDistance  call on to this one (passing 0 for radius) ?

**Comment by richardkchapman** [comment]
@ghalliday Couple of minor suggestions

File: ecllibrary/std/Str.ecl (hunk 1 above)
Comment (ghalliday) [reply ecllibrary/std/Str.ecl]: changed, and in uni.ecl

File: plugins/unicodelib/unicodelib.cpp (hunk 2 above)
Comment (ghalliday) [reply plugins/unicodelib/unicodelib.cpp]: changed

**Comment by ghalliday** [comment]
@richardkchapman both changes made.  Worth waiting for smoke test as a sanity check.

**Pull request merged**